Automattic/woocommerce-payments
# Update node to v16 (LTS) and npm to v8

Follow up on #3849

#### Changes proposed in this Pull Request
- Update node to `v16.17.0` (latest LTS) and npm to `v8.15.0` which matches that node version.
- Update `engines` entry on `package.json` and `.nvmrc` file.
- Add `.npmrc` with:
  - `legacy-peer-deps=true` to allow us to use v8 until we start updating our dependencies and fixing peer deps issues.
  - `engine-strict=true` to prevent using older versions that may cause issues, and get a proper error showing which one you must use.
- Use [overrides](https://docs.npmjs.com/cli/v8/configuring-npm/package-json) to fix `@automattic/puppeteer-utils` issues, loading [a fixed branch version](https://github.com/Automattic/puppeteer-utils/tree/update/babel-deps) instead. Until there is another solution (p1662463533761059-slack-C01BWDDTGKX)
- Replace any reference to `@automattic/puppeteer-utils` with `@woocommerce/e2e-environment` which includes it.
- Recreate `package-lock.json` in v2.  Ensuring every dependency use exact version to make it consistent with our current approach.
- Update failing tests to use `Object. defineProperty` to mock window properties without errors.
- Update `@wordpress/data` from `v4.22.1` to `v4.23.1` to fix tests errors due to unhandled promise rejections on node +v15.

#### Testing instructions
- Check that the changes make sense.
- Running `npm ci` without the required versions should fail and show which ones are needed.
- Running `npm ci` with the required versions should finish without errors.
- Running `npm run test:e2e -- --showConfig` should output the same configuration.

-------------------


**Post merge**
